src/main/java/org/truffleruby/options/CommandLineException.java

```diff
@@ -26,56 +26,23 @@
  ***** END LICENSE BLOCK *****/
 package org.truffleruby.options;
 
-public class MainExitException extends RuntimeException {
-    private static final long serialVersionUID = -8585821821150293755L;
-    boolean usageError;
-    int status;
-    private boolean aborted;
-
-    public MainExitException(int status) {
-        this(status, true);
-    }
-
-    public MainExitException(int status, String message) {
-        super(message);
-
-        this.status = status;
-    }
-
-    public MainExitException(int status, boolean aborted) {
-        super("aborted");
-
-        this.status = status;
-        this.aborted = aborted;
-    }
+public class CommandLineException extends Exception {
 
-    public int getStatus() {
-        return status;
-    }
+    private static final long serialVersionUID = -8585821821150293755L;
 
-    public void setStatus(int status) {
-        this.status = status;
-    }
+    private final boolean usageError;
 
-    public void setAborted(boolean aborted) {
-        this.aborted = aborted;
+    public CommandLineException(String message) {
+        this(message, false);
     }
 
-    public boolean isAborted() {
-        return this.aborted;
+    public CommandLineException(String message, boolean usageError) {
+        super(message);
+        this.usageError = usageError;
     }
 
     public boolean isUsageError() {
```

Comment:
I might rename it `shouldShowUsage` so it doesn't seem like an imperative. But both seem like a question about what you should do with the error - which is up to you not the exception. `isUsageError` is a question of fact about the error itself.